### PR TITLE
VID improvements

### LINKED ISSUE #355: Verify VID commitment when fetching payload or common data from another query service

### PR DESCRIPTION
**Pull request by jbearer**
* Handle views 0 and 1, so we no longer have the edge case where VID data is missing for these views
* Fix tests accordingly
* When fetching VID payload or common data from another query service, verify results against expected commitment
* Use standardized VID constructor from HotShot

Closes #355

### CODE REVIEW

File: src/fetching/provider/query_service.rs [hunk 1]
```diff
@@ -43,15 +48,47 @@ where
...
+                ))
+                .send()
+        );
+        match res {
+            Ok((payload, common)) => {
+                // Verify that the data we retrieved is consistent with the request we made.
+                let num_storage_nodes = VidSchemeType::get_num_storage_nodes(common.common());
+                let bytes = match payload.data().encode() {
+                    Ok(bytes) => bytes.collect::<Vec<_>>(),
+                    Err(err) => {
```
[the remark this answers is not in the document]
Comment (jbearer) [reply src/fetching/provider/query_service.rs:71]: Yeah, could probably clean this up a bit by having it return `anyhow::Error` and just doing the `tracing` at the top level. Not a priority for now though
